### PR TITLE
Added dto validation

### PR DESCRIPTION
**Pull request by artemklishch**

### CODE REVIEW

File: src/main/java/org/example/intro/dto/CreateBookDto.java [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
-        String title,
-        String author,
-        BigDecimal price,
-        String description,
-        String isbn,
-        String coverImage
-) {
+@Data
+public class CreateBookDto {
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/intro/dto/CreateBookDto.java]:
```suggestion
    @NotBlank
```

for String, check everywhere

File: src/main/java/org/example/intro/dto/CreateBookDto.java [hunk 2]
```diff
@@ -1,13 +1,26 @@
...
+    String title;
+
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/intro/dto/CreateBookDto.java]: remove empty lines between fields

File: src/main/java/org/example/intro/dto/CreateBookDto.java [hunk 3]
```diff
@@ -1,13 +1,26 @@
...
+    @NotNull
+    String author;
+
+    @NotNull
+    @Min(0)
+    BigDecimal price;
+
+    String description;
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/intro/dto/CreateBookDto.java:23]: Validate size

File: src/main/java/org/example/intro/dto/CreateBookDto.java [hunk 4]
```diff
@@ -1,13 +1,26 @@
...
+
+    @NotNull
+    String isbn;
+
+    String coverImage;
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/intro/dto/CreateBookDto.java:27]: Validate size

File: src/main/java/org/example/intro/controller/BookController.java [hunk 5]
```diff
@@ -23,7 +22,8 @@ public List<BookDto> findAll() {
     }
 
     @PostMapping
-    public BookDto create(@RequestBody CreateBookDto requestDto) {
+    @ResponseStatus(HttpStatus.CREATED)
+    public BookDto create(@RequestBody @Valid CreateBookDto requestDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/intro/controller/BookController.java:26]: Validate dto in search endpoint

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution

File: src/main/java/org/example/intro/dto/CreateBookDto.java (hunk 4 above)
Comment (artemklishch) [reply src/main/java/org/example/intro/dto/CreateBookDto.java:27]: ++, updated, i didn't know what size to set and set min value in 10 charachters

File: src/main/java/org/example/intro/dto/CreateBookDto.java (hunk 3 above)
Comment (artemklishch) [reply src/main/java/org/example/intro/dto/CreateBookDto.java:23]: ++, updated.

File: src/main/java/org/example/intro/dto/CreateBookDto.java (hunk 2 above)
Comment (artemklishch) [reply src/main/java/org/example/intro/dto/CreateBookDto.java]: ++, updated.

File: src/main/java/org/example/intro/dto/CreateBookDto.java (hunk 1 above)
Comment (artemklishch) [reply src/main/java/org/example/intro/dto/CreateBookDto.java]: ++, thank you. updated.

File: src/main/java/org/example/intro/controller/BookController.java (hunk 5 above)
Comment (artemklishch) [reply src/main/java/org/example/intro/controller/BookController.java:26]: Unfortunately i haven't figured out how to validate parameters here. I saw that we can validate incoming data in the request body. But this is GET query. 
The only thing i had found - is validating the type of request parameter and added additional method to the "CustomGlobalExceptionHandler" class.
If i can improve it - could you please help me with that? what certainly to make here? any resources where i would be able to take that information.
Thank you!

Comment (Elena-Bruyako) [reply src/main/java/org/example/intro/controller/BookController.java:26]: My mistake, don't need to validate

**Review by Elena-Bruyako — APPROVED** [review]
approve

**Pull request merged**